pkg/pit/basecamp.go

```diff
@@ -166,7 +166,7 @@ var basecampGlobalString = `{
 	"kubernetes-weave-mtu": "1376",
 	"cilium-operator-replicas": "1",
 	"k8s-primary-cni": "weave",
-	"cilium-kube-proxy-replacement": "strict",
+	"cilium-kube-proxy-replacement": "disabled",
```

Comment:
Does this have any backward compatibility issues with 1.4 and 1.5?
e.g. if someone runs this copy of CSI in a 1.5 environment, would it have side effects?

Reply:
No, the `cilium-kube-proxy-replacement` value isn't used as we don't deploy Cilium in either release.

The default primary CNI is still Weave and this particular attribute is only used if Cilium is installed.

https://github.com/Cray-HPE/node-images/blob/main/boxes/ncn-node-images/kubernetes/files/scripts/common/apply-networking-manifests.sh#L55-L61

Reply:
Ok.
If it was used, I was going to suggest using the CSM versioning package in CSI to wrap the usage of `basecampGlobalString`.

```diff
diff --git a/pkg/pit/basecamp.go b/pkg/pit/basecamp.go
index b50ee852..3203276b 100644
--- a/pkg/pit/basecamp.go
+++ b/pkg/pit/basecamp.go
@@ -28,6 +28,7 @@ import (
        "bytes"
        "encoding/json"
        "fmt"
+       "github.com/Cray-HPE/cray-site-init/pkg/csm"
        "log"
        "regexp"
        "strings"
@@ -260,6 +284,10 @@ func MakeBasecampGlobals(v *viper.Viper, logicalNcns []csi.LogicalNCN, shastaNet
        if err != nil {
                return global, err
        }
+       _, eval := csm.CompareMajorMinor("1.6")
+       if eval < 0 {
+               global["cilium-kube-proxy-replacement"] = "strict"
+       }
 
        // First loop through and see if there's a viper flag
        // We register a few aliases because flags don't necessarily match data.json keys

```